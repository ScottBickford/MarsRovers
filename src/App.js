import React from "react";
import "./App.css";
import { Container } from "semantic-ui-react";
import RoverInstructionsPage from "./features/roverInstructions/RoverInstructionsPage";

function App() {
  return (
    <Container className='main'>
      <h2>Rover Instructions</h2>
      <RoverInstructionsPage />
    </Container>
  );
}

export default App;
