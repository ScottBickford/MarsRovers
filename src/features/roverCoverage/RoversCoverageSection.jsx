import React, { Fragment } from 'react'
import { Grid } from 'semantic-ui-react';
import { RoverCoverage } from './RoverCoverage';
import { RoverTotalCoverage } from './RoverTotalCoverage';

export const RoversCoverageSection = ({roverMovements}) => {
  return (
    <Fragment>
    <h3>Visual Coverage</h3>
    <Grid
      columns={
        roverMovements.upperRightCoords.x <= 15
          ? 3
          : roverMovements.upperRightCoords.x <= 23
          ? 2
          : 1
      }
      stackable
    >
      {roverMovements.rovers.map(rover => {
        return (
          <RoverCoverage
            key={rover.id}
            upperRightCoords={roverMovements.upperRightCoords}
            rover={rover}
          />
        );
      })}
    </Grid>
    <RoverTotalCoverage
      roverMovements={roverMovements}
    />
  </Fragment>
  )
}
