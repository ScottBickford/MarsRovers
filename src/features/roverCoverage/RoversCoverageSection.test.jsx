import React from 'react';
import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { RoversCoverageSection } from './RoversCoverageSection';
import { roverMovementsTestObject } from '../../App.test';

describe('RoverInstructionsOutput', () => {
  test('snapshot renders', () => {
    const component = renderer.create(<RoversCoverageSection roverMovements={roverMovementsTestObject} />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('passes all props to Output', () => {
    const { getByText } = render(<RoversCoverageSection roverMovements={roverMovementsTestObject} />);
    const buttonElement = getByText(/Visual Coverage/i);
    expect(buttonElement).toBeInTheDocument();
  });
});