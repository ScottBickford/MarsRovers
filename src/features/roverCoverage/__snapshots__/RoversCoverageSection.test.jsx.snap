// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RoverInstructionsOutput snapshot renders 1`] = `
Array [
  <h3>
    Visual Coverage
  </h3>,
  <div
    className="ui stackable three column grid"
  >
    <div
      className="column"
    >
      <label>
        Rover 
        1
         coverage
      </label>
      <table
        className="ui celled collapsing table"
      >
        <tbody
          className=""
        >
          <tr
            className=""
          >
            <td
              className=""
              style={
                Object {
                  "backgroundColor": "white",
                }
              }
            />
            <td
              className=""
              style={
                Object {
                  "backgroundColor": "white",
                }
              }
            />
            <td
              className=""
              style={
                Object {
                  "backgroundColor": "white",
                }
              }
            />
            <td
              className=""
              style={
                Object {
                  "backgroundColor": "white",
                }
              }
            />
            <td
              className=""
              style={
                Object {
                  "backgroundColor": "white",
                }
              }
            />
            <td
              className=""
              style={
                Object {
                  "backgroundColor": "white",
                }
              }
            />
          </tr>
          <tr
            className=""
          >
            <td
              className=""
              style={
                Object {
                  "backgroundColor": "white",
                }
              }
            />
            <td
              className=""
              style={
                Object {
                  "backgroundColor": "white",
                }
              }
            />
            <td
              className=""
              style={
                Object {
                  "backgroundColor": "white",
                }
              }
            />
            <td
              className=""
              style={
                Object {
                  "backgroundColor": "white",
                }
              }
            />
            <td
              className=""
              style={
                Object {
                  "backgroundColor": "white",
                }
              }
            />
            <td
              className=""
              style={
                Object {
                  "backgroundColor": "white",
                }
              }
            />
          </tr>
          <tr
            className=""
          >
            <td
              className=""
              style={
                Object {
                  "backgroundColor": "white",
                }
              }
            />
            <td
              className=""
              style={
                Object {
                  "backgroundColor": "lightblue",
                }
              }
            />
            <td
              className=""
              style={
                Object {
                  "backgroundColor": "white",
                }
              }
            />
            <td
              className=""
              style={
                Object {
                  "backgroundColor": "white",
                }
              }
            />
            <td
              className=""
              style={
                Object {
                  "backgroundColor": "white",
                }
              }
            />
            <td
              className=""
              style={
                Object {
                  "backgroundColor": "white",
                }
              }
            />
          </tr>
          <tr
            className=""
          >
            <td
              className=""
              style={
                Object {
                  "backgroundColor": "lightblue",
                }
              }
            />
            <td
              className=""
              style={
                Object {
                  "backgroundColor": "lightblue",
                }
              }
            />
            <td
              className=""
              style={
                Object {
                  "backgroundColor": "white",
                }
              }
            />
            <td
              className=""
              style={
                Object {
                  "backgroundColor": "white",
                }
              }
            />
            <td
              className=""
              style={
                Object {
                  "backgroundColor": "white",
                }
              }
            />
            <td
              className=""
              style={
                Object {
                  "backgroundColor": "white",
                }
              }
            />
          </tr>
          <tr
            className=""
          >
            <td
              className=""
              style={
                Object {
                  "backgroundColor": "lightblue",
                }
              }
            />
            <td
              className=""
              style={
                Object {
                  "backgroundColor": "lightblue",
                }
              }
            />
            <td
              className=""
              style={
                Object {
                  "backgroundColor": "white",
                }
              }
            />
            <td
              className=""
              style={
                Object {
                  "backgroundColor": "white",
                }
              }
            />
            <td
              className=""
              style={
                Object {
                  "backgroundColor": "white",
                }
              }
            />
            <td
              className=""
              style={
                Object {
                  "backgroundColor": "white",
                }
              }
            />
          </tr>
          <tr
            className=""
          >
            <td
              className=""
              style={
                Object {
                  "backgroundColor": "white",
                }
              }
            />
            <td
              className=""
              style={
                Object {
                  "backgroundColor": "white",
                }
              }
            />
            <td
              className=""
              style={
                Object {
                  "backgroundColor": "white",
                }
              }
            />
            <td
              className=""
              style={
                Object {
                  "backgroundColor": "white",
                }
              }
            />
            <td
              className=""
              style={
                Object {
                  "backgroundColor": "white",
                }
              }
            />
            <td
              className=""
              style={
                Object {
                  "backgroundColor": "white",
                }
              }
            />
          </tr>
        </tbody>
      </table>
    </div>
    <div
      className="column"
    >
      <label>
        Rover 
        2
         coverage
      </label>
      <table
        className="ui celled collapsing table"
      >
        <tbody
          className=""
        >
          <tr
            className=""
          >
            <td
              className=""
              style={
                Object {
                  "backgroundColor": "white",
                }
              }
            />
            <td
              className=""
              style={
                Object {
                  "backgroundColor": "white",
                }
              }
            />
            <td
              className=""
              style={
                Object {
                  "backgroundColor": "white",
                }
              }
            />
            <td
              className=""
              style={
                Object {
                  "backgroundColor": "white",
                }
              }
            />
            <td
              className=""
              style={
                Object {
                  "backgroundColor": "white",
                }
              }
            />
            <td
              className=""
              style={
                Object {
                  "backgroundColor": "white",
                }
              }
            />
          </tr>
          <tr
            className=""
          >
            <td
              className=""
              style={
                Object {
                  "backgroundColor": "white",
                }
              }
            />
            <td
              className=""
              style={
                Object {
                  "backgroundColor": "white",
                }
              }
            />
            <td
              className=""
              style={
                Object {
                  "backgroundColor": "white",
                }
              }
            />
            <td
              className=""
              style={
                Object {
                  "backgroundColor": "white",
                }
              }
            />
            <td
              className=""
              style={
                Object {
                  "backgroundColor": "white",
                }
              }
            />
            <td
              className=""
              style={
                Object {
                  "backgroundColor": "white",
                }
              }
            />
          </tr>
          <tr
            className=""
          >
            <td
              className=""
              style={
                Object {
                  "backgroundColor": "white",
                }
              }
            />
            <td
              className=""
              style={
                Object {
                  "backgroundColor": "white",
                }
              }
            />
            <td
              className=""
              style={
                Object {
                  "backgroundColor": "white",
                }
              }
            />
            <td
              className=""
              style={
                Object {
                  "backgroundColor": "lightblue",
                }
              }
            />
            <td
              className=""
              style={
                Object {
                  "backgroundColor": "lightblue",
                }
              }
            />
            <td
              className=""
              style={
                Object {
                  "backgroundColor": "lightblue",
                }
              }
            />
          </tr>
          <tr
            className=""
          >
            <td
              className=""
              style={
                Object {
                  "backgroundColor": "white",
                }
              }
            />
            <td
              className=""
              style={
                Object {
                  "backgroundColor": "white",
                }
              }
            />
            <td
              className=""
              style={
                Object {
                  "backgroundColor": "white",
                }
              }
            />
            <td
              className=""
              style={
                Object {
                  "backgroundColor": "white",
                }
              }
            />
            <td
              className=""
              style={
                Object {
                  "backgroundColor": "white",
                }
              }
            />
            <td
              className=""
              style={
                Object {
                  "backgroundColor": "lightblue",
                }
              }
            />
          </tr>
          <tr
            className=""
          >
            <td
              className=""
              style={
                Object {
                  "backgroundColor": "white",
                }
              }
            />
            <td
              className=""
              style={
                Object {
                  "backgroundColor": "white",
                }
              }
            />
            <td
              className=""
              style={
                Object {
                  "backgroundColor": "white",
                }
              }
            />
            <td
              className=""
              style={
                Object {
                  "backgroundColor": "white",
                }
              }
            />
            <td
              className=""
              style={
                Object {
                  "backgroundColor": "lightblue",
                }
              }
            />
            <td
              className=""
              style={
                Object {
                  "backgroundColor": "lightblue",
                }
              }
            />
          </tr>
          <tr
            className=""
          >
            <td
              className=""
              style={
                Object {
                  "backgroundColor": "white",
                }
              }
            />
            <td
              className=""
              style={
                Object {
                  "backgroundColor": "white",
                }
              }
            />
            <td
              className=""
              style={
                Object {
                  "backgroundColor": "white",
                }
              }
            />
            <td
              className=""
              style={
                Object {
                  "backgroundColor": "white",
                }
              }
            />
            <td
              className=""
              style={
                Object {
                  "backgroundColor": "white",
                }
              }
            />
            <td
              className=""
              style={
                Object {
                  "backgroundColor": "white",
                }
              }
            />
          </tr>
        </tbody>
      </table>
    </div>
  </div>,
  <div
    style={
      Object {
        "paddingTop": 15,
      }
    }
  >
    <label>
      Total Coverage
    </label>
    <table
      className="ui celled collapsing table"
    >
      <tbody
        className=""
      >
        <tr
          className=""
        >
          <td
            className=""
            style={
              Object {
                "backgroundColor": "white",
              }
            }
          />
          <td
            className=""
            style={
              Object {
                "backgroundColor": "white",
              }
            }
          />
          <td
            className=""
            style={
              Object {
                "backgroundColor": "white",
              }
            }
          />
          <td
            className=""
            style={
              Object {
                "backgroundColor": "white",
              }
            }
          />
          <td
            className=""
            style={
              Object {
                "backgroundColor": "white",
              }
            }
          />
          <td
            className=""
            style={
              Object {
                "backgroundColor": "white",
              }
            }
          />
        </tr>
        <tr
          className=""
        >
          <td
            className=""
            style={
              Object {
                "backgroundColor": "white",
              }
            }
          />
          <td
            className=""
            style={
              Object {
                "backgroundColor": "white",
              }
            }
          />
          <td
            className=""
            style={
              Object {
                "backgroundColor": "white",
              }
            }
          />
          <td
            className=""
            style={
              Object {
                "backgroundColor": "white",
              }
            }
          />
          <td
            className=""
            style={
              Object {
                "backgroundColor": "white",
              }
            }
          />
          <td
            className=""
            style={
              Object {
                "backgroundColor": "white",
              }
            }
          />
        </tr>
        <tr
          className=""
        >
          <td
            className=""
            style={
              Object {
                "backgroundColor": "white",
              }
            }
          />
          <td
            className=""
            style={
              Object {
                "backgroundColor": "lightblue",
              }
            }
          />
          <td
            className=""
            style={
              Object {
                "backgroundColor": "white",
              }
            }
          />
          <td
            className=""
            style={
              Object {
                "backgroundColor": "lightblue",
              }
            }
          />
          <td
            className=""
            style={
              Object {
                "backgroundColor": "lightblue",
              }
            }
          />
          <td
            className=""
            style={
              Object {
                "backgroundColor": "lightblue",
              }
            }
          />
        </tr>
        <tr
          className=""
        >
          <td
            className=""
            style={
              Object {
                "backgroundColor": "lightblue",
              }
            }
          />
          <td
            className=""
            style={
              Object {
                "backgroundColor": "lightblue",
              }
            }
          />
          <td
            className=""
            style={
              Object {
                "backgroundColor": "white",
              }
            }
          />
          <td
            className=""
            style={
              Object {
                "backgroundColor": "white",
              }
            }
          />
          <td
            className=""
            style={
              Object {
                "backgroundColor": "white",
              }
            }
          />
          <td
            className=""
            style={
              Object {
                "backgroundColor": "lightblue",
              }
            }
          />
        </tr>
        <tr
          className=""
        >
          <td
            className=""
            style={
              Object {
                "backgroundColor": "lightblue",
              }
            }
          />
          <td
            className=""
            style={
              Object {
                "backgroundColor": "lightblue",
              }
            }
          />
          <td
            className=""
            style={
              Object {
                "backgroundColor": "white",
              }
            }
          />
          <td
            className=""
            style={
              Object {
                "backgroundColor": "white",
              }
            }
          />
          <td
            className=""
            style={
              Object {
                "backgroundColor": "lightblue",
              }
            }
          />
          <td
            className=""
            style={
              Object {
                "backgroundColor": "lightblue",
              }
            }
          />
        </tr>
        <tr
          className=""
        >
          <td
            className=""
            style={
              Object {
                "backgroundColor": "white",
              }
            }
          />
          <td
            className=""
            style={
              Object {
                "backgroundColor": "white",
              }
            }
          />
          <td
            className=""
            style={
              Object {
                "backgroundColor": "white",
              }
            }
          />
          <td
            className=""
            style={
              Object {
                "backgroundColor": "white",
              }
            }
          />
          <td
            className=""
            style={
              Object {
                "backgroundColor": "white",
              }
            }
          />
          <td
            className=""
            style={
              Object {
                "backgroundColor": "white",
              }
            }
          />
        </tr>
      </tbody>
    </table>
  </div>,
]
`;
